internal/service/warehouse_default.go

```diff
@@ -34,6 +41,26 @@ func (s *WarehouseDefault) Update(id int, warehouse models.WarehouseDTO) (w mode
 }
 
 func (s *WarehouseDefault) Delete(id int) (err error) {
+	allSections, err := s.repo.SectionsDB.GetAll()
+	if err != nil {
+		return
+	}
+	for _, section := range allSections {
+		if section.WarehouseID == id {
+			err = ErrWarehouseServiceSectionsAssociated
+			return
+		}
+	}
+	allEmployees, err := s.repo.EmployeesDB.GetAll()
+	if err != nil {
+		return
+	}
+	for _, employee := range allEmployees {
+		if employee.WarehouseID == id {
+			err = ErrWarehouseServiceEmployeesAssociated
+			return
+		}
+	}
```

Comment:
[discussão] no MR da @Izabelly-Melo , ela tinha feito essa validação com o GetById já pronto, não seria melhor do que repetir a lógica?

Reply:
Através do warehouses não tenho acesso aos ids associados. Por isso fiz o for loop. Em employees, é possível usar o GetById porque tem o campo warehouse_id.

Reply:
verdade, é pra checar se algum section ou algum employee tem um warehouse id, boa!